File: src/main/java/com/codeperfection/shipit/dto/transporter/CreateTransporterDto.java
package com.codeperfection.shipit.dto.transporter;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateTransporterDto {

    @NotBlank
    @Size(max = 256)
    private String name;

    @NotNull
    @Min(0)
    private Integer capacity;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/transporter/CreateTransporterDto.kt
package com.codeperfection.shipit.dto.transporter

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class CreateTransporterDto(
    @field:NotBlank
    @field:Size(max = 256)
    val name: String,
    @field:Min(1)
    val capacity: Int
)
